python/pyspark/sql/connect/dataframe.py

```diff
@@ -201,6 +202,34 @@ def sort(self, *cols: "ColumnOrString") -> "DataFrame":
         """Sort by a specific column"""
         return DataFrame.withPlan(plan.Sort(self._plan, *cols), session=self._session)
 
+    def sample(
+        self,
+        withReplacement: Optional[Union[float, bool]] = None,
+        fraction: Optional[Union[int, float]] = None,
+        seed: Optional[int] = None,
+    ) -> "DataFrame":
+        from pyspark.sql import DataFrame as PySparkDataFrame
+
+        (withReplacement, fraction, seed) = PySparkDataFrame._prepare_augments_for_sample(
+            withReplacement, fraction, seed
+        )
+        if withReplacement is None:
+            withReplacement = False
+        if seed is None:
+            # TODO: make 'seed' optional in proto, then we can use 'Utils.random.nextLong' in JVM
```

Comment:
@amaliujia We should really consider this. The principle is to move code implementation to the server side as much as possible. We just moved the identifier parsing logic to server side, and we should probably do the same for parameter default values.

Reply:
This makes sense.

@zhengruifeng I am thinking you can wrap this `seed` into a proto message and in that case the server side can know if this is set or not? In that case, the server side can does the random generation.

Reply:
This is an example: https://github.com/apache/spark/pull/38275

Reply:
yeah, let me make this change